### DISCUSSION TITLE
Capacitor v3 plugins not working when live reload

### DISCUSSION THREAD

**Discussion by Elvincth** [ionic-team/capacitor#4937] [Q&A]
I have tried the following code with ```@capacitor/app``` plugin, when it is live reload mode it will said  ```: App plugin is not implemented on android```, but if I turned off live reload mode it work fine.

```
  useEffect(() => {
    if (Capacitor.isNativePlatform() && !LIVE_RELOAD && document) {
      App.addListener("backButton", (e) => {
        // Use of location.pathname is also correct
        if (window.location.pathname === "/") {
          App.exitApp();
        } else {
          router.back();
        }
      });
    }
  }, []); // eslint-disable-line
```

**Comment by jcesarmobile** [comment]
How are you running live reload?

**Reply by Elvincth** [reply]
https://capacitorjs.com/docs/guides/live-reload

**Reply by Elvincth** [reply]
Also, I used this as an starter: https://github.com/mlynch/nextjs-tailwind-ionic-capacitor-starter.

**Reply by Elvincth** [reply]
I only got the not implemented error when live reload is enabled, once I turned it off and build the apk again the plugin work fine, so it seems that in cap3 plugins are not working when live reload mode turned on.

my config 
```
server = {
    url: "192.168.128.73:3000",
    cleartext: true,
}
  ```

**Reply by jcesarmobile** [reply]
I've been able to reproduce randomly on that app, sometimes it takes a lot of time to load and the loaded url is `192.168.1.129:3000`, in that case it doesn't work, but sometimes it's `192.168.1.129:3000/tabs/feed` and it works.

Since it's taking a lot of time to load, the code that injects capacitor is timing out and not injecting the needed js because of that and that makes plugins to not work, but in the end, the problem is the timeout of the server. Make sure the dev server has started and the url is accesible before running the Capacitor app.

**Reply by Elvincth** [reply]
Got it thanks 🙏
